### PR TITLE
Synth surgery changes

### PR DESCRIPTION
**Pull request by Shroopy**
## About The Pull Request
Adds synthetic versions of many surgeries. Because synths don't heal the same way organics do, surgical steps for them now have to be done in reverse to close the surgery. Liquid solder also now heals all organs in synths.
WIP: 
Make roboticists the best at synth surgeries by checking for huds/wire knowledge?

## Why It's Good For The Game
No more organic surgeries on synths, makes synth surgery more interesting than organic surgery with different tools.

## Proof Of Testing
WIP

## Changelog

:cl:
add: Added new surgeries for synthetic humanoids, all of which are self-doable.
del: Stopped organic surgeries working on synths.
add: Synth surgeries now have to be fully reversed. Robots can't put themselves back together!
add: Liquid solder also now all organs in synths, not just their brains.
del: Synths no longer get negative mood from surgery.
/:cl:

### CODE REVIEW

File: code/modules/surgery/organ_manipulation.dm
Comment (Majkl-J) [inline code/modules/surgery/organ_manipulation.dm:1]: Changes to this file can be made modular

File: code/datums/components/surgery_initiator.dm
Comment (Majkl-J) [inline code/datums/components/surgery_initiator.dm:139]:
```suggestion
			patient.balloon_alert(user, "need a [is_robotic ? "screwdriver": "cautery"] in an inactive slot to stop the surgery!") // BUBBER EDIT
```

This can probably be brough up to tg but eh works too

File: modular_skyrat/modules/synths/code/surgery/mechanic_steps.dm
Comment (Majkl-J) [inline modular_skyrat/modules/synths/code/surgery/mechanic_steps.dm:65]: This can also be made modular

File: modular_skyrat/modules/synths/code/surgery/robot_brain_surgery.dm
Comment (Majkl-J) [inline modular_skyrat/modules/synths/code/surgery/robot_brain_surgery.dm:1]: Also can be made modular

File: modular_skyrat/modules/synths/code/surgery/robot_chassis_restoration.dm
Comment (Majkl-J) [inline modular_skyrat/modules/synths/code/surgery/robot_chassis_restoration.dm:1]: Modularity

File: modular_skyrat/modules/synths/code/surgery/robot_healing.dm
Comment (Majkl-J) [inline modular_skyrat/modules/synths/code/surgery/robot_healing.dm:1]: Missing edit comments on this file

File: modular_skyrat/modules/synths/code/surgery/robot_mending.dm
Comment (Majkl-J) [inline modular_skyrat/modules/synths/code/surgery/robot_mending.dm:1]: Move to the modular folder

**Review by Majkl-J — CHANGES_REQUESTED** [review]

**Comment by Majkl-J** [comment]
wrong button

**Comment by StrangeWeirdKitten** [comment]
Synth chads rise up

But seriously, I do enjoy this as a potential leveraging away the pains of playing synthetic. My one off topic mention is that organ damage has utterly zero regeneration and is a massive pain in the ass for both robotics and the robot. Making liquid solder repair it would be another thing I approve of you want to tack that in or make a separate PR.

**Comment by Shroopy** [comment]
Really sorry about the delay

**Comment by Majkl-J** [comment]
> Really sorry about the delay

It's alright, you're pulling great work either way. Take your time

**Comment by Shroopy** [comment]
Oh no the modularization did not work right

**Comment by Shroopy** [comment]
Still messed up, oh no

**Comment by Shroopy** [comment]
Alright, things are finally looking good! I'm not sure if it's totally ready to be undrafted; I need a break to process the work I've done and make sure I'm not missing anything. But I would really appreciate a code review.

**Comment by Shroopy** [comment]
I've updated the PR description with what's been done so far and what's still WIP.

**Comment by xXPawnStarrXx** [comment]
I made a similar PR upstream on Skyrat because someone there requested it.https://github.com/Skyrat-SS13/Skyrat-tg/pull/26342#issuecomment-1926053357

File: code/datums/components/surgery_initiator.dm
Comment (nikothedude) [inline code/datums/components/surgery_initiator.dm:114]: maybe mark these as bubber edits?

**Comment by nikothedude** [comment]
please modularly edit the modified wounds descriptions to include that they can be surgically fixed

**Comment by nikothedude** [comment]
a simple desc += stuff would work

File: modular_zubbers/code/modules/synths/code/robot_mending.dm
Comment (nikothedude) [inline modular_zubbers/code/modules/synths/code/robot_mending.dm:167]: 20 what? use SECONDS define please

**Comment by nikothedude** [comment]
also, is there any particular reason surgeries were added to blunt wounds? i also noticed these surgeries are more forgiving than the non-surgical treatments, is there any reason for this? a notable difference is the lack of burn wound applied during the treatment of collapsed superstructure

i notice the blunt surgeries lack chance/speed mults for knowing robo/engi huds. this was originally implemented to encourage people to go to robo over med, because without the mults it allows med doctors to get a few easy to access tools and take over robos job. this is also why things like bone gel was added, because i recognized that medical still needed a way to treat the blunt wounds consistantly if robo was unavailable 

i also feel, at least in terms of blunt wounds, it makes treatment less unique and in fact more in line with standard treatment, which isnt a bad thing (i designed these wounds with some parity in mind), but we lose some flavorful things, such as using an RCD, or literally reforming the metal with your hands

overall i understand if synth wounds are overtuned and needs a nerf/tweak, but i wish i was consulted on this earlier

File: modular_zubbers/code/modules/synths/code/robot_mending.dm
Comment (Shroopy) [reply modular_zubbers/code/modules/synths/code/robot_mending.dm:167]: Original code I pulled this from did not have seconds define but will add

Comment (nevimer) [reply modular_zubbers/code/modules/synths/code/robot_mending.dm:167]: 20 means 2 seconds

**Comment by Shroopy** [comment]
@nikothedude 
I wanted to add non-damaging ways of treating synthetic blunt wounds. If you create more damage or even more wounds during treatment, that makes treatment that much harder. Organics have a choice to avoid that by doing surgery, but synths don't currently have that option, so I wanted to open up that option. My T2 surgery could probably just be commented out, as after looking at the nonsurgical treatment more, I don't think a surgery option is needed. 
I'll make sure to have requirements/benefits for using diagnostic huds, that's a great point.

**Review by Majkl-J — CHANGES_REQUESTED** [review]
`/datum/surgery_step/reattach_plating` often collides with organ manipulation or other similar surgeries, making them not closeable

**Comment by Shroopy** [comment]
I don't play on Bubberstation anymore, so I'm not going to be working on this anymore

**Comment by Majkl-J** [comment]
image

**Comment by Shroopy** [comment]
We're back

**Comment by Majkl-J** [comment]
This has to be on hold until next upstream merge

**Comment by Shroopy** [comment]
That's odd, that shouldn't be possible I think, I'll check it out

**Comment by Shroopy** [comment]
@Erol509 I'm not able to reproduce that

**Comment by Erol509** [comment]
> @Erol509 I'm not able to reproduce that

By using drapes again on the patient.

**Comment by Erol509** [comment]
Nevermind, on your pr it seems to not be there

**Comment by Shroopy** [comment]
I had the hardware manip surgery pop up for the head for an organic with no head, I'll look into it

**Comment by Shroopy** [comment]
I think this fixed it

**Comment by StrangeWeirdKitten** [comment]
@Shroopy is this PR done? I am planning a merge for it.

**Comment by Shroopy** [comment]
Sorry for the late reply. Yes, this is finished, barring anything that has come up during the testmerge that I haven't heard about.

**Comment by StrangeWeirdKitten** [comment]
Sorry this has been taking a bit. This is basically going to be our biggest priority once we cut down on the test merges.

**Comment by KeptheKobold** [comment]
The disassemble option for synths isn't showing up for synthetic humanoids and fully augmented podpersons, with or without the surgery table. On round 5481.
Sorry if this isn't the right place to post issues!

**Comment by plsleavemealon** [comment]
If this is presently on live it is horrifically bugged, we also cannot do brain surgery on androids or any augmented persons.  Unsure if this is the case for synths or not.

The naming schema is also very confusing, and leaves us guessing at what does what.

**Comment by Doom-external** [comment]
Another little bug to report, quite a few surgeries are missing off augmented people. I had a fully augged (originally podperson) come to me for surgery and I was unable to amputate limbs to replace them with new ones.

**Comment by KeptheKobold** [comment]
The system shock dampening surgery doesn't seem to do anything. Both self surgery and being performed by someone else it does nothing to stop you from getting shocked, or reduce the amount you get shocked to my eyes. Though i dont know much about the organic counterpart.

**Comment by Spoodersspooders** [comment]
I don't know too much about the circumstances behind it, but I recently ran into an issue regarding being unable to reattach a limb someone lost, neither jamming it in nor draping their missing arm for surgery worked. The drape just gave the message that there wasn't any surgery to do. They were an anthromorph and a synthetic, if it's any help.

**Comment by Shroopy** [comment]
https://github.com/Bubberstation/Bubberstation/discussions/2106
Please place bug reports here, and I am working on the issues.

**Comment by StrangeWeirdKitten** [comment]
Tell me when you're ready for another TM

**Comment by nikothedude** [comment]
honestly id be nerfing them more if skyrat wasnt a literally dead repo (fork when)

**Comment by Majkl-J** [comment]
Once you update this Ill run it through one last review

**Comment by Majkl-J** [comment]
Staling this out again

**Comment by Shroopy** [comment]
> Tell me when you're ready for another TM

Ready for another TM, please unstale. @Majkl-J

**Pull request closed without merging**
